src/modules/builtin/exit.rs

```diff
@@ -0,0 +1,48 @@
+use heraclitus_compiler::prelude::*;
+use crate::modules::expression::expr::Expr;
+use crate::docs::module::DocumentationModule;
+use crate::modules::types::{Type, Typed};
+use crate::translate::module::TranslateModule;
+use crate::utils::{ParserMetadata, TranslateMetadata};
+
+#[derive(Debug, Clone)]
+pub struct Exit {
+    code: Expr
+}
+
+impl SyntaxModule<ParserMetadata> for Exit {
+    syntax_name!("Exit");
+
+    fn new() -> Self {
+        Exit {
+            code: Expr::new()
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        token(meta, "exit")?;
+        syntax(meta, &mut self.code)?;
```

Comment:
Just thinking if we should allow omitting this parameter - this way we could just exit with zero exit code.

```
exit
// Same as `exit 0`
```

Reply:
It's kinda too late tbh but updated the PR to handle parsing of the code argument for the Exit statement as optional. If the code is omitted during parsing, it will now default to 0.